x-pack/plugins/security_solution/public/common/components/exceptions/exceptionable_endpoint_event_fields.json

```diff
@@ -0,0 +1,131 @@
+[
+  "Endpoint.policy.applied.id",
```

Comment:
I just copied all of them from x-pack/plugins/security_solution/public/common/components/exceptions/exceptionable_endpoint_fields.json and removed some ones. 
But yes, agree we should define this list.
cc: @kevinlog

Reply:
Let's have @ferullo @gabriellandau take a look at this file and we can adjust as needed.

following from https://github.com/elastic/security-team/issues/959 it sounds like we're OK to allow most fields except `event.kind` that we specifically called out.

Does this look OK otherwise?

Reply:
So exceptions are for alerts, which don't quite overlap with events.  For example, `registry.path` will never appear in an alert, and hence never be exceptionlistable.  Here's a quick listing of all the flat event fields.  This explicitly excludes the `alerts` subdirectory, which can contain fields present in alerts but never events.  Thoughts?

```
C:\git\endpoint-package\schemas\v1>grep -hoRi "flat_name: .*$" file library metadata metadata network process registry security | sort | uniq
flat_name: '@timestamp'
flat_name: agent.id
flat_name: agent.name
flat_name: agent.type
flat_name: agent.version
flat_name: data_stream.dataset
flat_name: data_stream.namespace
flat_name: data_stream.type
flat_name: destination.address
flat_name: destination.bytes
flat_name: destination.domain
flat_name: destination.geo.city_name
flat_name: destination.geo.continent_name
flat_name: destination.geo.country_iso_code
flat_name: destination.geo.country_name
flat_name: destination.geo.location
flat_name: destination.geo.name
flat_name: destination.geo.region_iso_code
flat_name: destination.geo.region_name
flat_name: destination.ip
flat_name: destination.packets
flat_name: destination.port
flat_name: destination.registered_domain
flat_name: destination.top_level_domain
flat_name: dll.code_signature.exists
flat_name: dll.code_signature.status
flat_name: dll.code_signature.subject_name
flat_name: dll.code_signature.trusted
flat_name: dll.code_signature.valid
flat_name: dll.Ext
flat_name: dll.Ext.code_signature
flat_name: dll.Ext.code_signature.exists
flat_name: dll.Ext.code_signature.status
flat_name: dll.Ext.code_signature.subject_name
flat_name: dll.Ext.code_signature.trusted
flat_name: dll.Ext.code_signature.valid
flat_name: dll.Ext.load_index
flat_name: dll.hash.md5
flat_name: dll.hash.sha1
flat_name: dll.hash.sha256
flat_name: dll.hash.sha512
flat_name: dll.name
flat_name: dll.path
flat_name: dll.pe.company
flat_name: dll.pe.description
flat_name: dll.pe.file_version
flat_name: dll.pe.imphash
flat_name: dll.pe.original_file_name
flat_name: dll.pe.product
flat_name: dns.Ext
flat_name: dns.Ext.options
flat_name: dns.Ext.status
flat_name: dns.question.name
flat_name: dns.question.registered_domain
flat_name: dns.question.subdomain
flat_name: dns.question.top_level_domain
flat_name: dns.question.type
flat_name: dns.resolved_ip
flat_name: ecs.version
flat_name: elastic.agent
flat_name: elastic.agent.id
flat_name: Endpoint.policy
flat_name: Endpoint.policy.applied
flat_name: Endpoint.policy.applied.id
flat_name: Endpoint.policy.applied.name
flat_name: Endpoint.policy.applied.status
flat_name: Endpoint.status
flat_name: event.action
flat_name: event.category
flat_name: event.code
flat_name: event.created
flat_name: event.dataset
flat_name: event.Ext
flat_name: event.Ext.correlation
flat_name: event.Ext.correlation.id
flat_name: event.hash
flat_name: event.id
flat_name: event.ingested
flat_name: event.kind
flat_name: event.module
flat_name: event.outcome
flat_name: event.provider
flat_name: event.sequence
flat_name: event.severity
flat_name: event.type
flat_name: file.accessed
flat_name: file.attributes
flat_name: file.created
flat_name: file.ctime
flat_name: file.device
flat_name: file.directory
flat_name: file.drive_letter
flat_name: file.Ext
flat_name: file.Ext.code_signature
flat_name: file.Ext.code_signature.exists
flat_name: file.Ext.code_signature.status
flat_name: file.Ext.code_signature.subject_name
flat_name: file.Ext.code_signature.trusted
flat_name: file.Ext.code_signature.valid
flat_name: file.Ext.entropy
flat_name: file.Ext.header_data
flat_name: file.Ext.monotonic_id
flat_name: file.Ext.original
flat_name: file.Ext.original.gid
flat_name: file.Ext.original.group
flat_name: file.Ext.original.mode
flat_name: file.Ext.original.name
flat_name: file.Ext.original.owner
flat_name: file.Ext.original.path
flat_name: file.Ext.original.uid
flat_name: file.Ext.windows
flat_name: file.Ext.windows.zone_identifier
flat_name: file.extension
flat_name: file.gid
flat_name: file.group
flat_name: file.hash.md5
flat_name: file.hash.sha1
flat_name: file.hash.sha256
flat_name: file.hash.sha512
flat_name: file.inode
flat_name: file.mime_type
flat_name: file.mode
flat_name: file.mtime
flat_name: file.name
flat_name: file.owner
flat_name: file.path
flat_name: file.path.caseless
flat_name: file.path.text
flat_name: file.pe.company
flat_name: file.pe.description
flat_name: file.pe.file_version
flat_name: file.pe.imphash
flat_name: file.pe.original_file_name
flat_name: file.pe.product
flat_name: file.size
flat_name: file.target_path
flat_name: file.target_path.caseless
flat_name: file.target_path.text
flat_name: file.type
flat_name: file.uid
flat_name: group.domain
flat_name: group.Ext
flat_name: group.Ext.real
flat_name: group.Ext.real.id
flat_name: group.Ext.real.name
flat_name: group.id
flat_name: group.name
flat_name: host.architecture
flat_name: host.domain
flat_name: host.hostname
flat_name: host.id
flat_name: host.ip
flat_name: host.mac
flat_name: host.name
flat_name: host.os.Ext
flat_name: host.os.Ext.variant
flat_name: host.os.family
flat_name: host.os.full
flat_name: host.os.full.caseless
flat_name: host.os.full.text
flat_name: host.os.kernel
flat_name: host.os.name
flat_name: host.os.name.caseless
flat_name: host.os.name.text
flat_name: host.os.platform
flat_name: host.os.version
flat_name: host.type
flat_name: host.uptime
flat_name: http.request.body.bytes
flat_name: http.request.body.content
flat_name: http.request.body.content.text
flat_name: http.request.bytes
flat_name: http.response.body.bytes
flat_name: http.response.body.content
flat_name: http.response.body.content.text
flat_name: http.response.bytes
flat_name: http.response.Ext
flat_name: http.response.Ext.version
flat_name: http.response.status_code
flat_name: message
flat_name: network.bytes
flat_name: network.community_id
flat_name: network.direction
flat_name: network.iana_number
flat_name: network.packets
flat_name: network.protocol
flat_name: network.transport
flat_name: network.type
flat_name: package.name
flat_name: process.args
flat_name: process.args_count
flat_name: process.code_signature.exists
flat_name: process.code_signature.status
flat_name: process.code_signature.subject_name
flat_name: process.code_signature.trusted
flat_name: process.code_signature.valid
flat_name: process.command_line
flat_name: process.command_line.caseless
flat_name: process.command_line.text
flat_name: process.entity_id
flat_name: process.executable
flat_name: process.executable.caseless
flat_name: process.executable.text
flat_name: process.exit_code
flat_name: process.Ext
flat_name: process.Ext.ancestry
flat_name: process.Ext.authentication_id
flat_name: process.Ext.code_signature
flat_name: process.Ext.code_signature.exists
flat_name: process.Ext.code_signature.status
flat_name: process.Ext.code_signature.subject_name
flat_name: process.Ext.code_signature.trusted
flat_name: process.Ext.code_signature.valid
flat_name: process.Ext.defense_evasions
flat_name: process.Ext.session
flat_name: process.Ext.token.elevation
flat_name: process.Ext.token.elevation_type
flat_name: process.Ext.token.integrity_level_name
flat_name: process.hash.md5
flat_name: process.hash.sha1
flat_name: process.hash.sha256
flat_name: process.hash.sha512
flat_name: process.name
flat_name: process.name.caseless
flat_name: process.name.text
flat_name: process.parent.args
flat_name: process.parent.args_count
flat_name: process.parent.code_signature.exists
flat_name: process.parent.code_signature.status
flat_name: process.parent.code_signature.subject_name
flat_name: process.parent.code_signature.trusted
flat_name: process.parent.code_signature.valid
flat_name: process.parent.command_line
flat_name: process.parent.command_line.caseless
flat_name: process.parent.command_line.text
flat_name: process.parent.entity_id
flat_name: process.parent.executable
flat_name: process.parent.executable.caseless
flat_name: process.parent.executable.text
flat_name: process.parent.exit_code
flat_name: process.parent.Ext
flat_name: process.parent.Ext.code_signature
flat_name: process.parent.Ext.code_signature.exists
flat_name: process.parent.Ext.code_signature.status
flat_name: process.parent.Ext.code_signature.subject_name
flat_name: process.parent.Ext.code_signature.trusted
flat_name: process.parent.Ext.code_signature.valid
flat_name: process.parent.Ext.real
flat_name: process.parent.Ext.real.pid
flat_name: process.parent.hash.md5
flat_name: process.parent.hash.sha1
flat_name: process.parent.hash.sha256
flat_name: process.parent.hash.sha512
flat_name: process.parent.name
flat_name: process.parent.name.caseless
flat_name: process.parent.name.text
flat_name: process.parent.pe.company
flat_name: process.parent.pe.description
flat_name: process.parent.pe.file_version
flat_name: process.parent.pe.imphash
flat_name: process.parent.pe.original_file_name
flat_name: process.parent.pe.product
flat_name: process.parent.pgid
flat_name: process.parent.pid
flat_name: process.parent.ppid
flat_name: process.parent.thread.id
flat_name: process.parent.thread.name
flat_name: process.parent.title
flat_name: process.parent.title.text
flat_name: process.parent.uptime
flat_name: process.parent.working_directory
flat_name: process.parent.working_directory.caseless
flat_name: process.parent.working_directory.text
flat_name: process.pe.company
flat_name: process.pe.description
flat_name: process.pe.file_version
flat_name: process.pe.imphash
flat_name: process.pe.original_file_name
flat_name: process.pe.product
flat_name: process.pgid
flat_name: process.pid
flat_name: process.ppid
flat_name: process.thread.id
flat_name: process.thread.name
flat_name: process.title
flat_name: process.title.text
flat_name: process.uptime
flat_name: process.working_directory
flat_name: process.working_directory.caseless
flat_name: process.working_directory.text
flat_name: registry.data.bytes
flat_name: registry.data.strings
flat_name: registry.hive
flat_name: registry.key
flat_name: registry.path
flat_name: registry.value
flat_name: source.address
flat_name: source.bytes
flat_name: source.domain
flat_name: source.geo.city_name
flat_name: source.geo.continent_name
flat_name: source.geo.country_iso_code
flat_name: source.geo.country_name
flat_name: source.geo.location
flat_name: source.geo.name
flat_name: source.geo.region_iso_code
flat_name: source.geo.region_name
flat_name: source.ip
flat_name: source.packets
flat_name: source.port
flat_name: source.registered_domain
flat_name: source.top_level_domain
flat_name: user.domain
flat_name: user.email
flat_name: user.Ext
flat_name: user.Ext.real
flat_name: user.Ext.real.id
flat_name: user.Ext.real.name
flat_name: user.full_name
flat_name: user.full_name.text
flat_name: user.group.domain
flat_name: user.group.Ext
flat_name: user.group.Ext.real
flat_name: user.group.Ext.real.id
flat_name: user.group.Ext.real.name
flat_name: user.group.id
flat_name: user.group.name
flat_name: user.hash
flat_name: user.id
flat_name: user.name
flat_name: user.name.text
```

Reply:
@gabriellandau would it be OK to allow the above list, but just exclude `event.kind` ?

Reply:
I think so, plus `*.code_signature` if we're doing the `nested` wall-padding like we did for TrustedApps.

Reply:
talked with @gabriellandau offline.  Since we're not doing any special validations yet that Exceptions doesn't do, we'll leave the `*.code_signature` fields in place.

we'll still omit `event.kind` from this list.

This will make the available fields be this:

```
'@timestamp'
agent.id
agent.name
agent.type
agent.version
data_stream.dataset
data_stream.namespace
data_stream.type
destination.address
destination.bytes
destination.domain
destination.geo.city_name
destination.geo.continent_name
destination.geo.country_iso_code
destination.geo.country_name
destination.geo.location
destination.geo.name
destination.geo.region_iso_code
destination.geo.region_name
destination.ip
destination.packets
destination.port
destination.registered_domain
destination.top_level_domain
dll.code_signature.exists
dll.code_signature.status
dll.code_signature.subject_name
dll.code_signature.trusted
dll.code_signature.valid
dll.Ext
dll.Ext.code_signature
dll.Ext.code_signature.exists
dll.Ext.code_signature.status
dll.Ext.code_signature.subject_name
dll.Ext.code_signature.trusted
dll.Ext.code_signature.valid
dll.Ext.load_index
dll.hash.md5
dll.hash.sha1
dll.hash.sha256
dll.hash.sha512
dll.name
dll.path
dll.pe.company
dll.pe.description
dll.pe.file_version
dll.pe.imphash
dll.pe.original_file_name
dll.pe.product
dns.Ext
dns.Ext.options
dns.Ext.status
dns.question.name
dns.question.registered_domain
dns.question.subdomain
dns.question.top_level_domain
dns.question.type
dns.resolved_ip
ecs.version
elastic.agent
elastic.agent.id
Endpoint.policy
Endpoint.policy.applied
Endpoint.policy.applied.id
Endpoint.policy.applied.name
Endpoint.policy.applied.status
Endpoint.status
event.action
event.category
event.code
event.created
event.dataset
event.Ext
event.Ext.correlation
event.Ext.correlation.id
event.hash
event.id
event.ingested
event.module
event.outcome
event.provider
event.sequence
event.severity
event.type
file.accessed
file.attributes
file.created
file.ctime
file.device
file.directory
file.drive_letter
file.Ext
file.Ext.code_signature
file.Ext.code_signature.exists
file.Ext.code_signature.status
file.Ext.code_signature.subject_name
file.Ext.code_signature.trusted
file.Ext.code_signature.valid
file.Ext.entropy
file.Ext.header_data
file.Ext.monotonic_id
file.Ext.original
file.Ext.original.gid
file.Ext.original.group
file.Ext.original.mode
file.Ext.original.name
file.Ext.original.owner
file.Ext.original.path
file.Ext.original.uid
file.Ext.windows
file.Ext.windows.zone_identifier
file.extension
file.gid
file.group
file.hash.md5
file.hash.sha1
file.hash.sha256
file.hash.sha512
file.inode
file.mime_type
file.mode
file.mtime
file.name
file.owner
file.path
file.path.caseless
file.path.text
file.pe.company
file.pe.description
file.pe.file_version
file.pe.imphash
file.pe.original_file_name
file.pe.product
file.size
file.target_path
file.target_path.caseless
file.target_path.text
file.type
file.uid
group.domain
group.Ext
group.Ext.real
group.Ext.real.id
group.Ext.real.name
group.id
group.name
host.architecture
host.domain
host.hostname
host.id
host.ip
host.mac
host.name
host.os.Ext
host.os.Ext.variant
host.os.family
host.os.full
host.os.full.caseless
host.os.full.text
host.os.kernel
host.os.name
host.os.name.caseless
host.os.name.text
host.os.platform
host.os.version
host.type
host.uptime
http.request.body.bytes
http.request.body.content
http.request.body.content.text
http.request.bytes
http.response.body.bytes
http.response.body.content
http.response.body.content.text
http.response.bytes
http.response.Ext
http.response.Ext.version
http.response.status_code
message
network.bytes
network.community_id
network.direction
network.iana_number
network.packets
network.protocol
network.transport
network.type
package.name
process.args
process.args_count
process.code_signature.exists
process.code_signature.status
process.code_signature.subject_name
process.code_signature.trusted
process.code_signature.valid
process.command_line
process.command_line.caseless
process.command_line.text
process.entity_id
process.executable
process.executable.caseless
process.executable.text
process.exit_code
process.Ext
process.Ext.ancestry
process.Ext.authentication_id
process.Ext.code_signature
process.Ext.code_signature.exists
process.Ext.code_signature.status
process.Ext.code_signature.subject_name
process.Ext.code_signature.trusted
process.Ext.code_signature.valid
process.Ext.defense_evasions
process.Ext.session
process.Ext.token.elevation
process.Ext.token.elevation_type
process.Ext.token.integrity_level_name
process.hash.md5
process.hash.sha1
process.hash.sha256
process.hash.sha512
process.name
process.name.caseless
process.name.text
process.parent.args
process.parent.args_count
process.parent.code_signature.exists
process.parent.code_signature.status
process.parent.code_signature.subject_name
process.parent.code_signature.trusted
process.parent.code_signature.valid
process.parent.command_line
process.parent.command_line.caseless
process.parent.command_line.text
process.parent.entity_id
process.parent.executable
process.parent.executable.caseless
process.parent.executable.text
process.parent.exit_code
process.parent.Ext
process.parent.Ext.code_signature
process.parent.Ext.code_signature.exists
process.parent.Ext.code_signature.status
process.parent.Ext.code_signature.subject_name
process.parent.Ext.code_signature.trusted
process.parent.Ext.code_signature.valid
process.parent.Ext.real
process.parent.Ext.real.pid
process.parent.hash.md5
process.parent.hash.sha1
process.parent.hash.sha256
process.parent.hash.sha512
process.parent.name
process.parent.name.caseless
process.parent.name.text
process.parent.pe.company
process.parent.pe.description
process.parent.pe.file_version
process.parent.pe.imphash
process.parent.pe.original_file_name
process.parent.pe.product
process.parent.pgid
process.parent.pid
process.parent.ppid
process.parent.thread.id
process.parent.thread.name
process.parent.title
process.parent.title.text
process.parent.uptime
process.parent.working_directory
process.parent.working_directory.caseless
process.parent.working_directory.text
process.pe.company
process.pe.description
process.pe.file_version
process.pe.imphash
process.pe.original_file_name
process.pe.product
process.pgid
process.pid
process.ppid
process.thread.id
process.thread.name
process.title
process.title.text
process.uptime
process.working_directory
process.working_directory.caseless
process.working_directory.text
registry.data.bytes
registry.data.strings
registry.hive
registry.key
registry.path
registry.value
source.address
source.bytes
source.domain
source.geo.city_name
source.geo.continent_name
source.geo.country_iso_code
source.geo.country_name
source.geo.location
source.geo.name
source.geo.region_iso_code
source.geo.region_name
source.ip
source.packets
source.port
source.registered_domain
source.top_level_domain
user.domain
user.email
user.Ext
user.Ext.real
user.Ext.real.id
user.Ext.real.name
user.full_name
user.full_name.text
user.group.domain
user.group.Ext
user.group.Ext.real
user.group.Ext.real.id
user.group.Ext.real.name
user.group.id
user.group.name
user.hash
user.id
user.name
user.name.text
```

cc\ @dasansol92 @paul-tavares